pkg/apis/rke.cattle.io/v1/bootstrap.go

```diff
@@ -5,40 +5,81 @@ import (
-	metav1.TypeMeta   `json:",inline"`
+	metav1.TypeMeta `json:",inline"`
+
+	// Metadata is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-	Spec              RKEBootstrapSpec   `json:"spec"`
-	Status            RKEBootstrapStatus `json:"status,omitempty"`
+
+	// Spec is the desired state of RKEBootstrap.
+	Spec RKEBootstrapSpec `json:"spec"`
+
+	// Status is the observed state of RKEBootstrap.
+	// +optional
+	Status RKEBootstrapStatus `json:"status,omitempty"`
 }
 
+// RKEBootstrapSpec defines the desired state of RKEBootstrap.
 type RKEBootstrapSpec struct {
+	// ClusterName refers to the name of the CAPI Cluster associated with this RKEBootstrap.
+	// +optional
 	ClusterName string `json:"clusterName,omitempty"`
 }
 
+// RKEBootstrapStatus defines the observed state of RKEBootstrap.
 type RKEBootstrapStatus struct {
-	// Ready indicates the BootstrapData field is ready to be consumed
+	// Ready indicates the BootstrapData field is ready to be consumed.
+	// +optional
 	Ready bool `json:"ready,omitempty"`
 
 	// DataSecretName is the name of the secret that stores the bootstrap data script.
 	// +optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
 	DataSecretName *string `json:"dataSecretName,omitempty"`
 }
 
 // +genclient
-// +kubebuilder:skipversion
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=rkebootstraptemplates,scope=Namespaced,categories=cluster-api
+// +kubebuilder:storageversion
+// +kubebuilder:metadata:labels={"cluster.x-k8s.io/v1beta1=v1","auth.cattle.io/cluster-indexed=true"}
 
+// RKEBootstrapTemplate is the schema for the rkebootstraptemplates API.
 type RKEBootstrapTemplate struct {
-	metav1.TypeMeta   `json:",inline"`
+	metav1.TypeMeta `json:",inline"`
+
+	// Metadata is the standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-	Spec              RKEBootstrapTemplateSpec `json:"spec" wrangler:"required"`
+
+	// Spec is the desired state of RKEBootstrapTemplate.
+	// +required
+	Spec RKEBootstrapTemplateSpec `json:"spec" wrangler:"required"`
 }
 
+// RKEBootstrapTemplateSpec defines the desired state of RKEBootstrapTemplate.
 type RKEBootstrapTemplateSpec struct {
-	ClusterName string       `json:"clusterName,omitempty"`
-	Template    RKEBootstrap `json:"template,omitempty" wrangler:"required"`
+	// ClusterName refers to the name of the CAPI Cluster associated with this RKEBootstrap.
+	// +optional
+	ClusterName string `json:"clusterName,omitempty"`
+
+	// Template defines the desired state of KubeadmConfigTemplate.
+	// +required
+	Template RKEBootstrap `json:"template" wrangler:"required"`
```

Comment:
Notes to reviewers:
The `omitempty` JSON tag has been removed to align with the upstream. 
Additionally, this field cannot be nil in practice, as the Wrangler tag enforces it as required.